python/BioSimSpace/Sandpit/Exscientia/Align/_squash.py

```diff
@@ -113,9 +113,29 @@ def _squash_molecule(molecule, explicit_dummies=False):
         return molecule
 
     if explicit_dummies:
-        # We make sure we use the same coordinates at both endstates.
+        # Get the common core atoms
+        atom_mapping0_common = _squashed_atom_mapping(
+            molecule,
+            is_lambda1=False,
+            explicit_dummies=explicit_dummies,
+            environment=False,
+            dummies=False,
+        )
+        atom_mapping1_common = _squashed_atom_mapping(
+            molecule,
+            is_lambda1=True,
+            explicit_dummies=explicit_dummies,
+            environment=False,
+            dummies=False,
+        )
+        assert set(atom_mapping0_common) == set(atom_mapping1_common)
```

Comment:
Not good to use assert in non-test cases.

Reply:
Asserts should be used for statements that should always be true and are not the user's fault right? This is the case here.